owncloud/helperFunctions.js

```diff
@@ -667,12 +721,12 @@ helpers.prototype._getAllFileInfo = function(path, pathToStore) {
 				}
 
 				if (fl === 0) {
-					var count = filesToPut.length;
-					filesToPut[count] = {};
-					filesToPut[count].path = baseAddr;
-					filesToPut[count].localPath = localPath; ////////
-					filesToPut[count].files = [ file ];
-					count++;
+					var count2 = filesToPut.length;
+					filesToPut[count2] = {};
```

Comment:
are you appending values to the `filesToPut` array this way ?
Usually using `filesToPut.push(value)` is better.